Improvement: Add @osdk/cli README (#92)
palantir/osdk-ts: packages/cli/README.md

| --uploadOnly | Upload new site version only without setting as live [boolean][default: false] |

Auto Version options

| Option         | Description                                                                                                                                                         |
| -------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| --autoVersion  | Enable auto versioning [string][choices: "git-describe"]                                                                                                            |
| --gitTagPrefix | Prefix to match git tags on when 'git-describe' auto versioning is used. If not provided, all tags are matched and the prefix 'v ' is stripped if present. [string] |

`--version` and `--autoVersion` are mutually exclusive and only one can be passed.

If `git-describe` is used for `--autoVersion`, the CLI will try to infer the version by running the `git describe` command with optionally `--match=<gitTagPrefix>` set if `--gitTagPrefix` is passed.

> #### `foundry.config.json` config file
> 
> This file contains project related configuration for Foundry website hosting. If a valid project file is present in the directory where the CLI is run, the CLI will automatically load the project file and depending on the present data, the CLI will be able to apply default values for otherwise required options.
> 
> An example of a full `foundry.config.json` is show below:

Reviewer: Some wording changes and thoughts on moving this to a top level section as it doesn't just apply to the deploy subcommand? We could put it directly after "Running locally" for most visibility or after the subcommand documentation for less visibility

```suggestion
## Project config file

A `foundry.config.json` file may be used in your project's root to simplify usage of the CLI by populating default values for commands. The current working directory will be traversed up to the root directory and the first `foundry.config.json` file if found will be used.

This file contains project related configuration for Foundry website hosting. If a valid project file is present in the directory where the CLI is run, the CLI will automatically load the project file and depending on the present data, the CLI will be able to apply default values for otherwise required options.

An example of a full `foundry.config.json` is shown below:
```

```json
{
  "foundryUrl": "https://example.palantirfoundry.com",
  "site": {
    "application": "ri.third-party-applications.main.application.1c66b352-4e00-40d2-995d-061c9d533ace",
    "directory": "./dist",
    "autoVersion": {
      "type": "git-describe",
      "tagPrefix": ""
    }
  }
}
```

### `version` subcommand

The version subcommand allows users to manage their site versions.

The version subcommand is not runnable without passing a specific subcommand i.e (`list` or `get`)

#### `list` subcommand

The list subcommand allows users to list their site versions

```
npx @osdk/cli site version list
```

#### `get` subcommand

> The list subcommand allows users to get their site live version

Reviewer: ```suggestion
The get subcommand allows users to get their live site version
```

```
npx @osdk/cli site version get
```

#### `set <version>` subcommand

> The list subcommand allows users to set their site live version

Reviewer: ```suggestion
The set subcommand allows users to set their live site version
```

```
npx @osdk/cli site version set 0.1.0
```

#### `unset` subcommand

> The list subcommand allows users to clear their site live version

Reviewer: ```suggestion
The unset subcommand allows users to clear their live site version
```

```
npx @osdk/cli site version unset
```

#### `delete <version>` subcommand

> The list subcommand allows users to delete a site version

Reviewer: ```suggestion
The delete subcommand allows users to delete a site version
```

```
npx @osdk/cli site version delete 0.1.0
```

## `typescript` subcommand

## `auth` subcommand

> 

Reviewer: Let's remove these empty sections given they're unstable

## Help and debugging

> To see all supported commands and their expected arguments, run your command with the --help flag

Reviewer: ```suggestion
To see all supported commands and their expected arguments, run a command with the --help flag
```

```
npx @osdk/cli site version --help
```

> And for additional information for debugging, you can use the --verbose flag

Reviewer: ```suggestion
For additional information for debugging, use the --verbose flag
```
